srml/support/src/dispatch.rs

```diff
@@ -67,38 +67,122 @@ pub trait Parameter: Codec + Clone + Eq {}
 #[cfg(not(feature = "std"))]
 impl<T> Parameter for T where T: Codec + Clone + Eq {}
 
-/// Declare a module struct and implement the dispatch logic.
+/// Declares a `Module` struct and a `Call` enum, which implements the dispatch logic.
 ///
-/// Usually used as follows:
+/// ## Declaration
 ///
+/// ```
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+/// 		fn my_function(origin, var: u64) -> Result {
+///				// Your implementation
+///				Ok(())
+/// 		}
+///
+///			// Public functions are easily accessible to other modules.
+///			pub fn my_public_function(origin) -> Result {
+///				Ok(())
+/// 		}
+///		}
```

Comment:
Regarding indentation, please look at the actual file and/or rustdoc for review. I think the PR file diff view isn't rendering properly, but the indentation is correct in VS code and when the HTML is rendered.